src/tests/presets/test_erc721.cairo

```diff
@@ -0,0 +1,1165 @@
+
+    dispatcher.set_approval_for_all(OPERATOR(), false);
+    assert_event_approval_for_all(dispatcher.contract_address, OWNER(), OPERATOR(), false);
+
+    assert(!dispatcher.is_approved_for_all(OWNER(), OPERATOR()), 'Approval not revoked correctly');
+}
+
+#[test]
+#[available_gas(20000000)]
+#[should_panic(expected: ('ERC721: self approval', 'ENTRYPOINT_FAILED'))]
+fn test_set_approval_for_all_owner_equal_operator_true() {
+    let dispatcher = setup_dispatcher();
+    testing::set_contract_address(OWNER());
+    dispatcher.set_approval_for_all(OWNER(), true);
+}
+
+#[test]
+#[available_gas(20000000)]
+#[should_panic(expected: ('ERC721: self approval', 'ENTRYPOINT_FAILED'))]
+fn test_set_approval_for_all_owner_equal_operator_false() {
+    let dispatcher = setup_dispatcher();
+    testing::set_contract_address(OWNER());
+    dispatcher.set_approval_for_all(OWNER(), false);
+}
+
+//
+// transfer_from & transferFrom
+//
+
+#[test]
+#[available_gas(20000000)]
+fn test_transfer_from_owner() {
+    let dispatcher = setup_dispatcher();
+    let token_id = TOKEN_1;
+    let owner = OWNER();
+    let recipient = RECIPIENT();
+
+    // set approval to check reset
+    testing::set_contract_address(OWNER());
+    dispatcher.approve(OTHER(), token_id);
+    utils::drop_event(dispatcher.contract_address);
+
+    assert_state_before_transfer(dispatcher, owner, recipient, token_id);
+    assert(dispatcher.get_approved(token_id) == OTHER(), 'Approval not implicitly reset');
+
+    testing::set_contract_address(owner);
+    dispatcher.transfer_from(owner, recipient, token_id);
+    assert_only_event_transfer(dispatcher.contract_address, owner, recipient, token_id);
+
+    assert_state_after_transfer(dispatcher, owner, recipient, token_id);
+}
+
+#[test]
+#[available_gas(20000000)]
+fn test_transferFrom_owner() {
+    let dispatcher = setup_dispatcher();
+    let token_id = TOKEN_1;
+    let owner = OWNER();
+    let recipient = RECIPIENT();
+    // set approval to check reset
+    testing::set_contract_address(owner);
+    dispatcher.approve(OTHER(), token_id);
+    utils::drop_event(dispatcher.contract_address);
+
+    assert_state_before_transfer(dispatcher, owner, recipient, token_id);
+    assert(dispatcher.get_approved(token_id) == OTHER(), 'Approval not implicitly reset');
+
+    dispatcher.transferFrom(owner, recipient, token_id);
+    assert_only_event_transfer(dispatcher.contract_address, owner, recipient, token_id);
+
+    assert_state_after_transfer(dispatcher, owner, recipient, token_id);
+}
+
+#[test]
+#[available_gas(20000000)]
+#[should_panic(expected: ('ERC721: wrong sender', 'ENTRYPOINT_FAILED'))]
+fn test_transfer_from_nonexistent() {
+    let dispatcher = setup_dispatcher();
+    dispatcher.transfer_from(ZERO(), RECIPIENT(), TOKEN_1);
+}
+
+#[test]
+#[available_gas(20000000)]
+#[should_panic(expected: ('ERC721: wrong sender', 'ENTRYPOINT_FAILED'))]
+fn test_transferFrom_nonexistent() {
```

Comment:
this test is not checking non-existent but not-owned, am i right? that's why it results in `wrong sender`

Reply:
Good catch! Fixed